From fe16594272a7b74a8c901f2ba72780538ca04934 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Thu, 23 May 2019 09:05:09 +0200
Subject: [PATCH] libsinsp: Fix linking when using system libraries
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

libsinsp needs to be linked to curl & tbb when using system libraries
as well.  Otherwise, the build fail with lots of undefined symbols.

sysdig-CLA-1.0-signed-off-by: Michał Górny <mgorny@gentoo.org>
---
 userspace/libsinsp/CMakeLists.txt | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/userspace/libsinsp/CMakeLists.txt b/userspace/libsinsp/CMakeLists.txt
index bc851319a..9ad978209 100644
--- a/userspace/libsinsp/CMakeLists.txt
+++ b/userspace/libsinsp/CMakeLists.txt
@@ -145,7 +145,9 @@ add_library(sinsp STATIC ${SINSP_SOURCES})
 
 target_link_libraries(sinsp 
 	scap
-	"${JSONCPP_LIB}")
+	"${CURL_LIBRARIES}"
+	"${JSONCPP_LIB}"
+	"${TBB_LIB}")
 
 if(USE_BUNDLED_LUAJIT)
 	add_dependencies(sinsp luajit)
@@ -157,13 +159,9 @@ if(NOT WIN32)
 	endif()
 	if(USE_BUNDLED_CURL)
 		add_dependencies(sinsp curl)
-		target_link_libraries(sinsp
-			"${CURL_LIBRARIES}")
 	endif()
 	if(USE_BUNDLED_TBB)
 		add_dependencies(sinsp tbb)
-		target_link_libraries(sinsp
-			"${TBB_LIB}")
 	endif()
 
 	if(NOT APPLE)
